### PR TITLE
Add specs for start, stop & executions for Container Apps Jobs

### PR DESCRIPTION
**Pull request by anandanthony**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [x] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json [hunk 1]
```diff
@@ -758,6 +902,75 @@
...
+      "type": "object",
+      "properties": {
+        "name": {
+          "$ref": "#/definitions/JobExecutionName"
+        },
+        "status": {
+          "description": "Job execution status.",
+          "type": "string"
+        },
+        "startTime": {
```
Comment (vinisoto) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json:921]: >startTime [](http://example.com/codeflow?start=9&length=9)

also add endTIme

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json [hunk 2]
```diff
@@ -430,9 +450,136 @@
         },
         "x-ms-examples": {
           "Terminate a Container Apps Job": {
-            "$ref": "./examples/Job_Stop.json"
+            "$ref": "./examples/Job__Stop_Execution.json"
```
Comment (vinisoto) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json:453]: >__ [](http://example.com/codeflow?start=35&length=2)

typo? (double underscore)

Comment (anandanthony) [reply specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json]: @vinisoto yes, it was a typo.. fixed it.

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/examples/Job_Stop_Multiple.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        },
+        {
+          "name": "jobExecution2"
+        }
+      ]
+    },
+    "api-version": "2022-11-01-preview"
+  },
+  "responses": {
+    "200": {},
```
Comment (trajkobal) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/examples/Job_Stop_Multiple.json:19]: For this there can be errors as well. e.g. some of the executions in the list were non existing. So we want to at least return a response with successfully stopped jobs and the ones that weren't successful

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json [hunk 4]
```diff
@@ -349,13 +349,25 @@
             "type": "string",
             "pattern": "^[-\\w\\._\\(\\)]+$"
           },
+          {
+            "name": "template",
+            "in": "body",
+            "description": "Properties used to create a container apps job",
```
Comment (trajkobal) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json:355]: Properties used to start a container apps job instance.

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json [hunk 5]
```diff
@@ -407,6 +419,14 @@
             "required": true,
             "type": "string",
             "pattern": "^[-\\w\\._\\(\\)]+$"
+          },
+          {
+            "name": "jobExecutionName",
+            "in": "path",
+            "description": "Name of the Container Apps Job's Execution.",
```
Comment (trajkobal) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json:426]: Job execution name.

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json [hunk 6]
```diff
@@ -430,9 +450,136 @@
...
+            "description": "List of all job executions that should be stopped.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/JobExecutionNamesCollection"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Container Apps Jobs terminated successfully."
```
Comment (trajkobal) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json:499]: We need to return a list of jobs and their termination status. Some in the list can be successful some may not be.
So something like this [{jobExecutionName: execution name, Status: success},{jobExecutionName: execution name, Status: not fount}]

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json [hunk 7]
```diff
@@ -430,9 +450,136 @@
...
-            "$ref": "./examples/Job_Stop.json"
+            "$ref": "./examples/Job_Stop_Execution.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        }
+      }
+    },
```
Comment (trajkobal) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json:461]: We should also add the stopAll endpoint here and implement it to reduce overhead.

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/examples/Job_Executions.json [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  "responses": {
+    "200": {
+      "headers": {},
+      "body": {
+        "value": [
+          {
+            "name": "testcontainerApp0-pjxhsye",
+            "status": "Running",
+            "startTime": "2023-02-13T20:37:30+00:00",
+            "template": {
```
Comment (tdaroly) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/examples/Job_Executions.json:17]: should we add `endTime`

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json (hunk 7 above)
Comment (anandanthony) [reply specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json:461]: @trajkobal can we reuse this endpoint? If the customer does not give a payload consisting of the job executions to be stopped, we stop all? I believe we would have to rename the endpoint to "StopAll" just so their is no confusion and the payload will make sense then, that if we have a payload passed we will only stop those job executions.?
OR 
we need a seperate endpoint /stopAll? The only concern with that is we would have 3 endpoints just for stop.

- /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/jobs/{jobName}/executions/{jobExecutionName}/stop
- /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/jobs/{jobName}/stop
- /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/jobs/{jobName}/stopAll

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/examples/Job_Stop_Multiple.json (hunk 3 above)
Comment (anandanthony) [reply specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/examples/Job_Stop_Multiple.json]: @trajkobal updated this, now we are returning a list of job executions for which a stop was requested, along with the status.

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json (hunk 7 above)
Comment (vinisoto) [reply specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json:461]: What would be the difference between these two in this case?:

/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/jobs/{jobName}/stop
/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/jobs/{jobName}/stopAll

I think it is clearer to have only `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/jobs/{jobName}/stop`. If a the payload is empty, we stop all executions; and if the payload contains a list of executions, we stop only that ilst.

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json [hunk 9]
```diff
@@ -349,13 +349,25 @@
             "type": "string",
             "pattern": "^[-\\w\\._\\(\\)]+$"
           },
+          {
+            "name": "template",
```
Comment (rkmanda) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json:353]: >name [](http://example.com/codeflow?start=13&length=4)

pl check the warnings here and make corrections - [https://github.com/Azure/azure-rest-api-specs/pull/22594/checks?check_run_id=11428332921](https://github.com/Azure/azure-rest-api-specs/pull/22594/checks?check_run_id=11428332921)

Comment (anandanthony) [reply specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json:353]: @rkmanda we have addressed some of the comments. For the others related to the operation names, the current names best define the operation behavior. Can you please proceed with merging this PR?

**Review by tdaroly — APPROVED** [review]

**Pull request merged**